test/test_broker.hpp

```diff
@@ -649,6 +649,24 @@ class test_broker {
         sessions_.clear();
     }
 
+    void clear_all_retained_topics() {
+        retains_.clear();
+    }
+
+    template<typename Callback>
+    void property_visitor(MQTT_NS::v5::properties props, Callback&& callback) {
+        for (auto const& p : props) {
+            MQTT_NS::visit(
+                MQTT_NS::make_lambda_visitor(
+                    MQTT_NS::force_move(callback),
```

Comment:
force_move is bad.
For the forwarding reference std::forward<Callback> should be used.

However, I think that it is still wrong.
make_lambda_visitor is called multiple times. Because it is in the for loop.
The first call, callback is moved, the second call, accessing moved from object.